### DISCUSSION TITLE
Table names

### DISCUSSION THREAD

**Discussion by glewe** [codeigniter4/shield#733] [Ideas]
I like Shield and assume it will further grow and develop. It might be a picky request but considering that the Shield tables might reside in larger databases with many more tables I suggest to rename the two tables linking users to groups and permissions into

- auth_users_groups
- auth_users_permissions

This makes much more sense since the first term should refer to the primary object of the relation ('users'), specifically since it is an actual table in the database while groups and permissions are not.

I know that I can rename the tables in **app/Config/Auth.php** but I think it should be done in the source to be more consistent with common naming conventions.

**Comment by lonnieezell** [comment]
Thanks for your comment. 

I've also seen naming conventions state that it should be done alphabetically, which is currently done. I'm not sure if either is more "proper" and don't think that an additional migration to change the table name for almost 30,000 installs is something that's worth pursuing at this point. 

I look forward to seeing other ideas from you, though .

**Comment by glewe** [comment]
Hi Lonnie,
yes, you are right. This is really not important. Conventions and habits differ.
Thanks for taking the time to respond anyway.
Best regards,
George
